### ISSUE TITLE
Dataset normalization issue

### ISSUE THREAD

**Issue by zhourongleiden** [wengwenchao123/DDGCRN#4]
Thanks for your good work. In the code, the raw data is normalized by the mean and std values calculated from the whole dataset. This can cause an information leakage from the testing part and may be unfair for other baselines. A better choice could be calculating the mean and std using only the training and validation parts.

**Comment by wengwenchao123** [comment]
Thank you for your interest in this work. In fact, I had identified this issue earlier and made modifications to the code, enabling it to normalize the data using the mean and variance of the training set. However, after conducting tests, I found that the outcomes of both versions were identical. Consequently, I uploaded the original code to GitHub.

Regarding this result, I speculate that even though I normalized the original data using the mean and standard deviation values of the entire dataset, during training, my model learned only the data distribution within the training set. The data distribution in the test and validation sets, which the model did not learn from the training set, could not affect the final outcome.

But you are right, research requires meticulous attention. I will update the relevant code later, as you suggested. However, I am currently preparing for my PhD application, so I might not have the time to address this until the end of the year. Therefore, please be patient and wait for a while.

**Comment by zhourongleiden** [comment]
Thanks for the quick reply. Hope everything goes fine with your PhD application.
